src/shared/parse-helpers.c

```diff
@@ -196,3 +202,129 @@ int parse_socket_bind_item(
         *port_min = mn;
         return 0;
 }
+
+int open_file_parse(const char *v, OpenFile **ret) {
+        _cleanup_free_ char *options = NULL;
+        _cleanup_(open_file_freep) OpenFile *of = NULL;
+        int r;
+
+        assert(v);
+        assert(ret);
+
+        of = new0(OpenFile, 1);
+        if (!of)
+                return -ENOMEM;
+
+        r = extract_many_words(&v, ":", EXTRACT_DONT_COALESCE_SEPARATORS|EXTRACT_CUNESCAPE, &of->path, &of->fdname, &options, NULL);
+        if (r < 0)
+                return r;
+
+        if (r == 0)
+                return -EINVAL;
+
+        /* Enforce that at most 3 colon-separated words are present */
+        if (!isempty(v))
+                return -EINVAL;
+
+        for (const char *p = options;;) {
+                OpenFileFlags flag;
+                _cleanup_free_ char *word = NULL;
+
+                r = extract_first_word(&p, &word, ",", 0);
+                if (r < 0)
+                        return r;
+
+                if (r == 0)
+                        break;
+
+                flag = open_file_flags_from_string(word);
+                if (flag < 0)
+                        return flag;
+
+                if ((flag & ~_OPENFILE_MASK_PUBLIC) != 0)
+                        return -EINVAL;
+
+                of->flags |= flag;
+        }
+
+        r = open_file_check(of);
+        if (r < 0)
+                return r;
+
+        if (isempty(of->fdname)) {
+                free(of->fdname);
+                r = path_extract_filename(of->path, &of->fdname);
+                if (r < 0)
+                        return r;
+        } else if (!fdname_is_valid(of->fdname))
+                return -EINVAL;
```

Comment:
please do the fdnae_is_valid() check also if we extract the fname from the path for it.

Reply:
maybe move the fdname_is_valid() check into open_file_check()